pennylane/gradients/vjp.py

```diff
@@ -345,7 +346,7 @@ def vjp(tape, dy, gradient_fn, gradient_kwargs=None):
         return [], lambda _, num=None: None
 
     try:
-        if qml.math.allclose(dy, 0):
+        if not isinstance(dy, tuple) and qml.math.allclose(dy, 0):

```

Comment:
I think we should also cover the case where `dy` is a tuple, but every element from it is `0`, as per the comment in the following 3 lines.

Reply:
Added a check where `dy` is a (nested) tuple and all elements are `0`.